services/worker/src/worker/job_runners/split/duckdb_index.py

```diff
@@ -80,6 +82,28 @@ def check_indexable(feature: FeatureType) -> None:
     return indexable_columns
 
 
+def get_delete_operations(
+    all_repo_files: set[str], split_names: set[str], config: str, index_file_location: str
+) -> list[CommitOperationDelete]:
+    only_config_files: set[str] = {
+        file for file in all_repo_files if re.compile(f"^({re.escape(config)})/").match(file)
+    }
+    pattern_only_config_splits_files = re.compile(
+        f"^({'|'.join(re.escape(f'{config}/{split_name}') for split_name in split_names)})/"
+    )
+
+    # delete old files from non existent splits
+    files_to_delete: set[str] = {
+        file
+        for file in only_config_files
+        # delete the obsolete index in the current split
+        if file == index_file_location
```

Comment:
btw: we don't need to remove this, since we will always create CommitOperationAdd with the new value. And huggingface_hub will thus ignore this CommitOperationDelete